UIforETW/PowerStatus.cpp

```diff
@@ -108,125 +108,179 @@ void CPowerStatusMonitor::SampleBatteryStat()
 	for (int deviceNum = 0; deviceNum < maxBatteries; deviceNum++)
 	{
 		SP_DEVICE_INTERFACE_DATA did = { sizeof(did) };
-
-		if (SetupDiEnumDeviceInterfaces(hdev,
+		const BOOL enumDeviceInterfacesResult =
+			SetupDiEnumDeviceInterfaces(hdev,
 										0,
 										&GUID_DEVCLASS_BATTERY,
 										deviceNum,
-										&did))
+										&did);
+		if (!enumDeviceInterfacesResult)
+		{
+			const DWORD lastErr = ::GetLastError( );
+			if (lastErr == ERROR_NO_MORE_ITEMS)
+			{
+				break; // Enumeration failed - perhaps we're out of items
+			}
+			continue;
+		}
+
+		DWORD bytesNeeded = 0;
+		const BOOL getDetailResult = SetupDiGetDeviceInterfaceDetail(hdev,
+										&did,
+										0,
+										0,
+										&bytesNeeded,
+										0);
+
+		if (getDetailResult)
+		{
+			throw std::logic_error("Expected SetupDiGetDeviceInterfaceDetail failure!");

```

Comment:
I think use whichever is most compact. The expectation is that this will never occur so I'd hate to devote multiple lines of code to it. That means either std::terminate, or UIETWASSERT, I think.